test/suite/test_dump02.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# [TEST_TAGS]
+# wt_util
+# [END_TAGS]
+
+import wttest
+from suite_subprocess import suite_subprocess
+
+# test_dump02.py
+# Test the dump utility to find different keys.
+class test_dump(wttest.WiredTigerTestCase, suite_subprocess):
+    table_format = 'key_format=u,value_format=u'
+    uri = 'table:test_dump'
+    output = 'dump.out'
+    data_header = 'Data\n'
+
+    n_rows = 100
+
+    def gen_key(self, i):
+        return 'key' + str(i)
+
+    def gen_value(self, i):
+        return 'value' + str(i)
+    
+    def get_num_data_lines_from_dump(self, file):
+        """
+        Get the number of lines corresponding to data from a dump file.
+        """
+        lines = open(file).readlines()
+        data_started = False
```

Comment:
Purely optional alternate implementation for consideration
```suggestion
        data_start = lines.index(self.data_header)                      # This *will* throw if not found.
        assert self.data_header not in lines[data_start + 1:].     # Alternate to index to avoid ValueError exception.
        return len(lines) - data_start - 1
```

Reply:
Thanks! Updated in 35af2ecc685270f9ca7260df037b05052d662c8a